microsoft/utils/setup_aitl.py

```diff
@@ -0,0 +1,365 @@
+    is_actions_match = _compare_str_list(
+        exist_role["permissions"][0]["actions"], target_role["Actions"]
+    )
+    is_dataactions_match = _compare_str_list(
+        exist_role["permissions"][0]["dataActions"], target_role["DataActions"]
+    )
+    is_scopes_match = _compare_str_list(
+        exist_role["assignableScopes"], target_role["AssignableScopes"]
+    )
+
+    if (
+        (exist_role["description"] == target_role["Description"])
+        and (exist_role["roleName"] == target_role["Name"])
+        and is_actions_match
+        and is_dataactions_match
+        and is_scopes_match
+    ):
+        return True
+    else:
+        return False
+
+
+def _wait_role_propagate(
+    role_name: str, subscription_id: str, role_before_update: str = ""
+) -> None:
+    logging.info("In _wait_role_propagate: waiting for the role changes to propagate")
+
+    sleep_interval = 5
+    time_out = 400
+    required_continuous_change_count = 5
+    count = 0
+    start_time = time.time()
+    # check changes happened continuous multiple times to make sure cache is refreshed.
+    while time.time() - start_time < time_out:
+        logging.info("Checking change applied...")
+        get_role_cmd_result = _call_cmd(
+            f'az role definition list --name "{role_name}" --scope "{subscription_id}"'
+        )
+        role_after_update = get_role_cmd_result.stdout
+
+        # for role creation completed
+        if role_after_update.startswith("[]"):
+            count = 0
+            logging.info("--- No role found.")
+        else:
+            is_same = _compare_role_setting(role_before_update, role_after_update)
+            if is_same:
+                # if no changes then reset count 0
+                count = 0
+                logging.info("--- Role setting Not Updated.")
+            else:
+                count += 1
+                logging.info("--- Found changed results.")
+
+        if count >= required_continuous_change_count:
+            break
+        time.sleep(sleep_interval)
+
+    elapsed_time = time.time() - start_time
+    logging.info(f"Role is propagated in {elapsed_time} sec")
+
+
+def _init_arg_parser() -> Namespace:
+    parser = ArgumentParser(
+        description="Create/Update a custom role and grant to Service Principal"
+    )
+    parser.add_argument(
+        "-r",
+        "--role",
+        default="Azure Image Testing for Linux Delegator",
+        help="RoleName you want to create/update",
+    )
+    parser.add_argument(
+        "-p",
+        "--service_principal",
+        default="AzureImageTestingforLinux",
+        help="Service Principal Name you want to assign",
+    )
+    parser.add_argument(
+        "-s",
+        "--subscriptionId",
+        default="",
+        help="Subscription scope in Role assignment(default is current subscription)",
+    )
+
+    return parser.parse_args()
+
+
+def _set_target_role_parameters(
+    subscription_id: str, role_name: str
+) -> tuple[str, str]:
```

Comment:
Use `Tuple` from typing to be compatible with 3.8. You can see pipeline errors to fix it.